src/charm.py

```diff
@@ -71,6 +169,40 @@ def _on_config_changed(self, _):
                 value = self.model.config[key]
                 logger.info("Setting {} to: {}".format(key, value))
                 self._stored.config[key] = value
+
+        # this should go into install hook
+        uid = pwd.getpwnam("root").pw_uid
+        gid = grp.getgrnam("root").gr_gid
+
+        path = "/etc/nginx/ssl"
```

Comment:
**suggestion (non-blocking):** 

Move below logic to a single function can make it more easy for testing.
Also can isolate the side-effect.